### ISSUE TITLE
Additional Features

### ISSUE THREAD

**Issue by SteveImmanuel** [katsura-jp/pytorch-cosine-annealing-with-warmup#11]
Hi, I have some suggestions for features: 
- Can you add initial warmup variant so that the warmup steps only apply on the **first** cycle like the one shown [here](https://github.com/pytorch/pytorch/issues/80308)?
- Can you infer the `max_lr` parameter value from the optimizer base learning rate for each group? Currently, if you have an optimizer with multiple groups with different learning rates, all of their learning rate values will get overridden by the `max_lr`.

BTW very awesome implementation!

**Comment by katsura-jp** [comment]
Hi, @SteveImmanuel 

Thank you for your suggestions.
It looks so good and I am going to work on your suggestions.

Once I close the issue, I will add a new issue for each suggestion.

Thank you.
